src/execution.js

```diff
@@ -167,81 +167,99 @@ const getGeneratedMetricIdentifier = (item, aggregation, expressionCreator, hash
     return `${type}_${identifier}.generated.${prefix}${aggregation}.${hash}`;
 };
 
-const generatedMetricDefinition = item => {
-    const { title, format, sort } = item;
+const isDerived = measure => {
+    const type = get(measure, 'type');
+    return (type === 'fact' || type === 'attribute' || !allFiltersEmpty(measure));
+};
+
+const isDateCategory = ({ type }) => type === 'date';
+const isDateFilter = ({ dateFilter }) => dateFilter;
+
+const getCategories = ({ categories }) => map(categories, ({ category }) => category);
+const getFilters = ({ filters }) => filters;
```

Comment:
Just a small thing,  but the discrepancy between isDateCategory and isDateFilter might be confusing - the first accepts the inner object, while the second accepts the outer wrapper. I understand that in the case of filters you need to work with the outer wrappers.

You could however update isDateCategory and getCategories to correspond, eg.
```
const isDateCategory = ({ category }) => category.type === 'date';
const getCategories = ({ categories }) => categories
```
Then you'd need to update usage of getCategories correspondingly - maybe you don't really need getCategories and getFilters in that case.